<ShareButton /> mobile fixes

ElvisJG opened the pull request:
had to create a new branch based off main, rebranching did not fix the build issues

share button fix, mobile useEffect and cleanup state, small fix to TypeOfHelpInput missing key

ElvisJG commented:
> @ElvisJG can you give me more information about what i am supposed to see? I do not see the Share Your Results button that I was thinking this was for.

The Share button is currently turned off but when it's active again this will restrict the share function to screen widths less than 640px

admoorgit reviewed (changes requested):
Works great!
One tweak and we're good: either make it pop down underneath the zip earlier, or stick to the right side of the screen longer. It gets a little funky w/ positioning between 420 - 640px width:

[<img width="569" alt="image" src="https://user-images.githubusercontent.com/72033897/183918516-fe58a68f-4e88-452b-9bed-299aea34f852.png">](https://ownpath-pr-19.herokuapp.com/search?zip=80020&miles=5)

admoorgit commented:
@ElvisJG I just noticed — on the provider detail page, let's go with the same language and location as search results. So "Share link" in the upper right corner, on the same line as the "Back to search" link. Let me know if it'll be a big lift to move it.

ElvisJG commented:
> @ElvisJG I just noticed — on the provider detail page, let's go with the same language and location as search results. So "Share link" in the upper right corner, on the same line as the "Back to search" link. Let me know if it'll be a big lift to move it.

Done 👍 and the changes discussed in our convo

jlhogan commented on src/index.css:
```diff
@@ -174,3 +174,54 @@ code {
+  width: fit-content;
+}
+
+.shareButtonContainer {
```
instead of creating custom CSS classes in this index.css file, we're trying to use [styled-components](https://styled-components.com/). Let me know if you want to grab time to discuss how to implement that, but you should be able to just define a `ShareButtonContainer` component at the top of ShareButton like:
```
const ShareButtonContainer = styled.div`
  position: absolute;
  right: 0;
  top: 47%;
  transform: translate(0%, -50%);
`
```
and then use it as a normal react component

ElvisJG replied:
Done 👍

jlhogan commented on src/translations/en/pages.json:
```diff
@@ -29,10 +29,10 @@
     "fullDetailButton": "Full detail about this location",
     "mapHelper": " Tap a marker to pull up information for that location. To zoom in or out on the map on a mobile device, use a two finger pinching gesture.",
-    "share": "Share these results"
+    "share": "Share link"
```
content changes need to be made to the spanish translations as well

jlhogan reviewed (commented):
LGTM except all the custom CSS classes. let's use https://[styled-components](https://styled-components.com/).com/ instead. let me know if you want to discuss!

admoorgit reviewed (changes requested):
Close! 3 changes and I think we're good:
1. We need to set the break point a little earlier on the Spanish site, since the longer link text crashes into the other content (see screenshot)
2. "Link Copied!" text needs to be in Spanish. I did google translate and asked someone at BW who knows Spanish and I think we're good to go with "¡Enlace copiado!" 
3. I know we set the "Change" link to right aligned at smaller widths @ElvisJG but now that I'm seeing it in action, I think we should go back to having it parked just to the left of the zip code. The issue lies at certain widths where it replaces the share text location — which is a huge oversight on my part and is super confusing. Will a lot of people see this? Probably not. But I think we'll also want to adjust the zip code H1 on the spanish site anyway to be on 2 lines (I'll make another ticket) where this'll make more sense. Please slack me if you have questions

img width="802" alt="image" src="https://user-images.githubusercontent.com/72033897/184377641-c31078ad-5e24-4632-999d-23438bd88f57.png">

ElvisJG commented:
> 

1. Added a Spanish container for the content that breaks at 1040px width
2. Added "Link copied!" and "¡Enlace copiado!" to the translation files and applied those changes to the tooltip
3. Reverted the change 😄

admoorgit commented:
Looks good to me!

jlhogan committed:
Bring back the share button!

jlhogan committed:
actually works

cathydeng commented on src/components/ShareButton.tsx:
```diff
@@ -48,9 +58,36 @@ function ShareButton({ text }: ShareButtonProps) {
+  right: 0;
+  top: 47%;
+  transform: translate(0%, -50%);
+  @media screen and (max-width: 65em) {
```
nit: my hunch is that there's a DRY-er and more elegant way to write this if the only diff here is the max-width? but not a blocker

cathydeng commented on src/pages/Search/Search.tsx:
```diff
@@ -164,9 +191,17 @@ function Search() {
                       t("change")
                     )}
                   </Button>
+                  {i18n.language === "en" ? (
```
it's pretty awk to have this container logic outside of the`ShareButton` component - if we need to know the language, it'd be cleaner to keep this logic in the `ShareButton` component and have a prop for `lang`, and just export `ShareButton`

cathydeng reviewed (changes requested):
<img width="245" alt="Screen Shot 2022-08-16 at 10 53 27 AM" src="https://user-images.githubusercontent.com/1406537/184946385-9575b220-8a63-4b66-ba08-807b42212e48.png">

the positioning of the share button on the detail page is a lil wacky - i'm assuming this is not intentional and it's supposed to be on the right side of the page?

cathydeng commented on src/pages/ResultDetail.tsx:
```diff
@@ -51,17 +51,17 @@ function ResultDetail() {
   return (
     <GridContainer className="ResultDetail">
-      <div className="margin-y-2">
+      <div className="margin-y-2 display-flex flex-justify-between">
```
i think replacing `flex-justify-between` w/ `flex-justify` should fix alignment

jlhogan replied:
yes yup ty ty

jlhogan committed:
PR feedback

cathydeng reviewed (approved):
sorry i meant just exporting ShareButton and handling all the container stuff within the ShareButton component instead of needing to export both ShareButtonContainer and ShareButton. but LGTM and non-blocking, can clean up later

amdarnell7 reviewed (approved):
Works as expected, thank you

jlhogan commented:
@cathydeng RE 

> sorry i meant just exporting ShareButton and handling all the container stuff within the ShareButton component instead of needing to export both ShareButtonContainer and ShareButton. but LGTM and non-blocking, can clean up later

the share button needs to container in the Search page but is used vanilla in the result detail page so didn't want to require it however i am sure there are cleaner ways to do this but agree it's fine for now
